homeassistant/components/travel_time/__init__.py

```diff
@@ -0,0 +1,193 @@
+    ATTR_DISTANCE,
+    ATTR_DURATION,
+    ATTR_DURATION_IN_TRAFFIC,
+    ATTR_MODE,
+    ATTR_ORIGIN,
+    ATTR_ORIGIN_NAME,
+    ATTR_ROUTE,
+    ATTR_ROUTE_MODE,
+    ATTR_TRAFFIC_MODE,
+    ATTR_UNIT_SYSTEM,
+    DOMAIN,
+    UNIT_OF_MEASUREMENT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Track states and offer events for sensors."""
+    component = hass.data[DOMAIN] = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL
+    )
+
+    await component.async_setup(config)
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a config entry."""
+    return await hass.data[DOMAIN].async_setup_entry(entry)
+
+
+async def async_unload_entry(hass, entry):
+    """Unload a config entry."""
+    return await hass.data[DOMAIN].async_unload_entry(entry)
+
+
+class TravelTimeEntity(Entity):
+    """Representation of a travel_time entity."""
+
+    @property
+    def attribution(self) -> str:
+        """Get the attribution of the travel_time entity."""
+        return None
+
+    @property
+    def destination(self) -> str:
+        """Get the destination coordinates of the travel_time entity."""
+        return None
+
+    @property
+    def destination_name(self) -> str:
+        """Get the destination name of the travel_time entity."""
+        return None
+
+    @property
+    def distance(self) -> str:
+        """Get the distance of the travel_time entity."""
+        return None
+
+    @property
+    def duration(self) -> str:
+        """Get the duration without traffic of the travel_time entity."""
+        return None
+
+    @property
+    def duration_in_traffic(self) -> str:
+        """Get the duration with traffic of the travel_time entity."""
+        return None
+
+    @property
+    def icon(self) -> str:
+        """Icon to use in the frontend."""
+        return None
+
+    @property
+    def mode(self) -> str:
+        """Get the mode of travelling e.g car for this entity."""
+        return None
+
+    @property
+    def name(self) -> str:
+        """Get the name of the travel_time entity."""
+        return None
+
+    @property
+    def origin(self) -> str:
+        """Get the origin coordinates of the travel_time entity."""
+        return None
+
+    @property
+    def origin_name(self) -> str:
+        """Get the origin name of the travel_time entity."""
+        return None
+
+    @property
+    def route(self) -> str:
+        """Get the route of the travel_time entity."""
+        return None
+
+    @property
+    def route_mode(self) -> str:
+        """Get the route_mode e.g fastest of the travel_time entity."""
+        return None
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return the state of the travel_time entity."""
+        return None
```

Comment:
We should always have the base class define the state property. It should return one of the other properties.